hindsight/output/telemetry_s3_webrtc.lua

```diff
@@ -0,0 +1,229 @@
+if compression and compression ~= "zst" and compression ~= "gz" then
+    error("compression must be nil, zst or gz")
+end
+
+
+local function get_fqfn(path)
+    return string.format("%s/%s", batch_path, path)
+end
+
+
+local function close_fh(entry)
+    if not entry[2] then return end
+    entry[2]:close()
+    entry[2] = nil
+    fh_cnt = fh_cnt - 1
+end
+
+
+local function copy_file(path, entry)
+    close_fh(entry)
+    local t = os.time()
+    local cmd
+    if t == time_t then
+        buffer_cnt = buffer_cnt + 1
+    else
+        time_t = t
+        buffer_cnt = 0
+    end
+
+    local src  = get_fqfn(path)
+    local dim_path = string.gsub(path, "+", "/")
+    if compression == "zst" then
+        cmd = string.format("zstd -c %s | aws s3 cp - %s/%s/%d_%d_%s.%s", src,
+                            s3_path, dim_path, time_t, buffer_cnt, hostname, compression)
+    elseif compression == "gz" then
+        cmd = string.format("gzip -c %s | aws s3 cp - %s/%s/%d_%d_%s.%s", src,
+                            s3_path, dim_path, time_t, buffer_cnt, hostname, compression)
+    else
+        cmd = string.format("aws s3 cp %s %s/%s/%d_%d_%s", src,
+                            s3_path, dim_path, time_t, buffer_cnt, hostname)
+    end
+
+    print(cmd)
+    local ret = os.execute(cmd)
+    if ret ~= 0 then
+        return string.format("ret: %d, cmd: %s", ret, cmd)
+    end
+    files[path] = nil
+
+    local ok, err = os.remove(src);
+    if not ok then
+        return string.format("os.remove('%s') failed: %s", path, err)
+    end
+end
+
+
+local function get_entry(path)
+    local ct = os.time()
+    local t = files[path]
+    if not t then
+        t = {ct, nil} -- last active, file handle
+        files[path] = t
+    else
+        t[1] = ct
+    end
+
+    if not t[2] then
+        if max_file_handles ~= 0 then
+            if fh_cnt >= max_file_handles then
+                local oldest = ct + 60
+                local entry
+                for k,v in pairs(files) do -- if we max out file handles a lot we will want to make this more efficient
+                    local et = v[1]
+                    if v[2] and et < oldest then
+                        entry = v
+                        oldest = et
+                    end
+                end
+                if entry then close_fh(entry) end
+            end
+        end
+        t[2] = assert(io.open(get_fqfn(path), "a"))
+        fh_cnt = fh_cnt + 1
+    end
+    return t
+end
+
+local function check_payload (payload)
+    if type(payload) ~= "table" then return false end
+    local w = payload["webrtc"] or {}
+    local i = w["IceCandidatesStats"] or {}
+    if next(i["webrtc"] or {}) or next(i["loop"] or {}) then
+        return true
+    end
+    return false
+end
+
+local dimensions = ts3.validate_dimensions(read_config("dimension_file"))
+-- create the batch directory if it does not exist
+os.execute(string.format("mkdir -p %s", batch_path))
+
+function process_message()
```

Comment:
There is a lot of code duplication with telemetry_s3.lua; the common code should be factored out into a module.